tree/ntuple/v7/test/ntuple_storage_daos.cxx

```diff
@@ -3,23 +3,55 @@
 #include "ROOT/TestSupport.hxx"
 #include <iostream>
 
-TEST(RPageStorageDaos, Basics)
-{
-   ROOT::TestSupport::CheckDiagsRAII diags;
-   diags.optionalDiag(kWarning, "in int daos_init()", "This RNTuple build uses libdaos_mock. Use only for testing!");
-   diags.requiredDiag(kWarning, "ROOT::Experimental::Detail::RPageSinkDaos::RPageSinkDaos",
-                      "The DAOS backend is experimental and still under development.", false);
-   diags.requiredDiag(kWarning, "[ROOT.NTuple]", "Pre-release format version: RC 1", false);
+class RPageStorageDaos : public ::testing::Test {
+public:
+   std::unique_ptr<ROOT::TestSupport::CheckDiagsRAII> fTestRootDiags;
+
+   static constexpr const char fDaosContainerLabels[5][22] = {
+      "test-basics", "test-extended", "test-options", "test-multiple-ntuples", "test-caged-pages",
+   };
+
+   [[nodiscard]] static std::string GetDaosUri(int testIndex)
+   {
+      static const std::string testPoolUriPrefix("daos://" R__DAOS_TEST_POOL "/");
+      return {testPoolUriPrefix + fDaosContainerLabels[testIndex]};
+   }
+
+protected:
+   void SetUp() override
+   {
+      // Initialized at the start of each test to expect diagnostic messages from TestSupport
+      fTestRootDiags = std::make_unique<ROOT::TestSupport::CheckDiagsRAII>();
+      fTestRootDiags->optionalDiag(kWarning, "in int daos_init()",
+                                   "This RNTuple build uses libdaos_mock. Use only for testing!");
+      fTestRootDiags->requiredDiag(kWarning, "ROOT::Experimental::Detail::RPageSinkDaos::RPageSinkDaos",
+                                   "The DAOS backend is experimental and still under development.", false);
+      fTestRootDiags->requiredDiag(kWarning, "[ROOT.NTuple]", "Pre-release format version: RC 1", false);
+   }
 
-   std::string daosUri("daos://" R__DAOS_TEST_POOL "/container-test-1");
+   static void TearDownTestSuite()
+   {
+#if !R__DAOS_TEST_MOCK
+      const std::string sysCmd("daos cont destroy " R__DAOS_TEST_POOL " ");
+      for (const auto &label : fDaosContainerLabels) {
+         system((sysCmd + label).data());
+      }
+#endif
+   }
+};
 
+TEST_F(RPageStorageDaos, Basics)
+{
+   std::string daosUri = GetDaosUri(0);
+   const std::string ntupleName("ntuple");
```

Comment:
Although the string is small (and thus should be subject to Small-Buffer Optimization in `std::string`), I would use `std::string_view` instead (also in all the occurrences below).